mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -3242,24 +3246,42 @@ LogicalResult LLVM::ConstantOp::verify() {
     return emitOpError() << "does not support target extension type.";
   }
 
+  // Check that an attribute whose element type has floating point semantics
+  // `attributeFloatSemantics` is compatible with a type whose element type
+  // is `constantElementType`.
+  //
+  // Requirement is that either
+  // 1) They have identical floating point types.
+  // 2) `constantElementType` is an integer type of the same width as the float
+  //     attribute. This is to support builtin MLIR float types without LLVM
+  //     equivalents, see comments in getLLVMConstant for more details.
+  auto verifyFloatSemantics =
+      [this](const llvm::fltSemantics &attributeFloatSemantics,
+             Type constantElementType) -> LogicalResult {
+    if (auto floatType = dyn_cast<FloatType>(constantElementType)) {
+      if (&floatType.getFloatSemantics() != &attributeFloatSemantics) {
+        return emitOpError()
+               << "attribute and type have different float semantics";
+      }
+      return success();
+    }
+    unsigned floatWidth = APFloat::getSizeInBits(attributeFloatSemantics);
+    if (isa<IntegerType>(constantElementType)) {
+      if (!constantElementType.isInteger(floatWidth)) {
+        return emitOpError() << "expected integer type of width " << floatWidth;
+      }
```

Comment:
```suggestion
      if (!constantElementType.isInteger(floatWidth)) 
        return emitOpError() << "expected integer type of width " << floatWidth;
```
nit: for single line ifs we usually drop the braces